### PR TITLE
Use common GCS functions from superclass

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -1757,6 +1757,94 @@ void GCS_MAVLINK::handle_common_message(mavlink_message_t *msg)
...
+        /* fall through */
+    case MAVLINK_MSG_ID_MISSION_ITEM_INT:
+        /* fall through */
+    case MAVLINK_MSG_ID_MISSION_REQUEST_INT:
+        /* fall through */
+    case MAVLINK_MSG_ID_MISSION_REQUEST:
+        /* fall through */
+    case MAVLINK_MSG_ID_MISSION_SET_CURRENT:
+        /* fall through */
+    case MAV_CMD_MISSION_START:
```
Comment (OXINARF) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: This seems wrong. Message ID will never be a command ID.

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -1757,6 +1757,94 @@ void GCS_MAVLINK::handle_common_message(mavlink_message_t *msg)
...
+    // GCS has sent us a mission item, store to EEPROM
+    case MAVLINK_MSG_ID_MISSION_ITEM:           // MAV ID: 39
+    {
+        if (handle_mission_item(msg, *_mission)) {
+            DataFlash_Class::instance()->Log_Write_EntireMission(*_mission);
+        }
+        break;
+    }
+
+    case MAVLINK_MSG_ID_MISSION_ITEM_INT:
```
Comment (OXINARF) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Any reason the code below is duplicated from above?

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: That's... odd.  Fixed.

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 2 above)
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: That's the way it came in from the donor library.  I did see it, but decided I was moving code, not fixing it :-)

I'll add a patch on top to remove the duplication.

**Comment by peterbarker** [comment]
@OXINARF I've moved the MISSION_ACK stuff up now.

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 2 above)
Comment (OXINARF) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: That's funny because I noticed this because I saw one of the vehicles not having the duplicated code 🙂

**Comment by OXINARF** [comment]
@peterbarker Do you mind if I squash the last 4 commits?

@magicrub Again, why dev call topic? This is basically a non-functional change...

**Comment by peterbarker** [comment]
@OXINARF Not at all - squash away!

**Comment by OXINARF** [comment]
As agreed, squashed and merged, thanks!

**Pull request closed without merging**